### PR TITLE
Tasks system improvements

### PR DESCRIPTION
**Pull request by PalmerAL**
For #747.

This PR includes a bunch of improvements to the tasks system, which should hopefully address a lot of the feedback from the [survey](https://forms.zohopublic.com/minbrowser/form/Taskssurvey/formperma/b6qx5QWqMx5SW6NSef3sbITA28WE65AujUBvXemZ8YE):

* **Different session restore behavior**: Instead of always restoring your previous session, Min will now ask you whether you want to return to your previous task or not:
tasks session restore

* **Task collapsing**: You can collapse tasks so they take up less space in the overlay. When a task is collapsed, you'll see a preview of the last tab you had opened in that task, so you can tell what the task is about (and depending on how old the task is, you may also see the date you last opened the task).
tasks collapsing

* **Undo delete**, which should make it harder to accidentally delete tasks. (The click area of the delete button has also been expanded, so it's easier to delete tasks when you want to).
task  delete undo

* **Favicons?** There's a checkbox to display icons next to tabs. I'm not sure whether this will make it easier or harder to quickly scan the tab list, so please try both ways and let me know which way you think is better.
task favicons

* **Keyboard shortcuts to switch tasks**: alt+ctrl+1 switches to the first task you have open, alt+ctrl+2 to the second task, etc.

And also: 
* The animation when switching tasks has been removed.
* Simplified the way search result pages are displayed in the tab list.

### CODE REVIEW

**Comment by PalmerAL** [comment]
Links to prebuilt binaries:

macOS: https://send.firefox.com/download/4de02f5a0ff3290d/#PLpQ2cKMsMZrH7NDPB-S2w
windows installer: https://send.firefox.com/download/9e8a0a0071baa3f3/#dHq5B5vXXyw9h93UPMgZkw
linux .deb: https://send.firefox.com/download/00b108da9317c6ed/#SyhRWxAZp0jmqXm_p-1hkQ

If you have any feedback or suggestions on how to improve this, please let me know! In particular, I'm interested in feedback on the session restore changes and showing favicons in the task list - do these features make it easier or harder for you to use tasks?

You can comment on this PR, or if you don't have a Github account, you can email me at 280953907a@zoho.com with comments.

**Comment by PalmerAL** [comment]
@wickdchromosome @zantyru @hrotkogabor @HenriDev @kittencoder @Bisquick69 @korchoon @bentearzz @Zetsuban Since you mentioned on the survey that you would be interested in testing the new tasks system, there are links to binaries in the comment above, please let me know what you think!

**Comment by dbfreem** [comment]
I'll start by saying I don't use the task manager much. I like all the features you added. Two things I would add. Indent the urls inside each task and maybe use a different font. This will help to determine where one task ends and another begins. The second thing is its not very intuative on how to add new urls to each task when you have multiple task, and then you have the new task at the bottom. 

See screenshot below. How would I add something to the first task? 

image

**Comment by zantyru** [comment]
I can't try on the linux build. The reason: unresolved dependency - libxssl1 (not libxss1, but libxssl1 - 'L'). I am seeing a solution...

**Comment by dbfreem** [comment]
I was looking through the issues and saw #355 and #478 are either of those candidates for this PR?

**Comment by dbfreem** [comment]
@zantyru I looked and libxssl1 is in the list of dependencies for the Linux Build. Did you try to run `npm run buildLinuux`  or running electron directly from the root of the repo with `node_modules/.bin/electron .`

**Comment by BenTearzz** [comment]
Just tried it out. I like the favicons, although, they could be a little bigger.
I think that indentation to each URL and perhaps bold text for the task title would look even better.

**Comment by swiftmas** [comment]
Gotta say i really love the favicons they are just the right size to not be distracting but help me with quick targeting of tabs i need.

 I do a lot of mixed tasks, more like workflows. So I will work on a problem and have the 2 stack overflow tabs, the webapp page itself, the amazon web portal page and the queue system page that i'm testing with all open in one task. so having the option to show favicons makes a huge difference when i need to jump back into a task or my workflow requires me to jump to one of my more immutable tasks (non-temporary) for info and then back again. Really aids multitasking and jumping between workflows as well. Pairing that with collapsible tasks. its gold. this update seriously helps my productivity. I can essentially store a workflow for a problem and come back to it. I have been labeling tasks with the ticket numbers and opening them when i work on the specific tickets. I also appreciate giving the option for each of these features. 

tl;dr - everything feels really good. I hope im not in the minority for the favicons lol.

**Comment by PalmerAL** [comment]
@zantyru I added `libxssl1` to the dependency list because of #571, but since it looks like that package doesn't actually exist, I guess it should probably be `libxss1`?

APT still installed it even though it complained about the missing dependency, so hopefully that should work for you?

**Comment by PalmerAL** [comment]
@dbfreem I think I want to finish this off first, but we should definitely consider adding those at some point.

**Comment by PalmerAL** [comment]
@dbfreem @BenTearzz @swiftmas Thanks for the feedback on the icons! Since it sounds like they're useful, I think I'll probably go ahead and keep them.

Originally, my intent was to put the favicons on the left, but it looks like they ended up on the right side on Windows and Linux. That's fixed now (d5415a2d419146d84ee780d436b8b039275b5874), hopefully it's an improvement:

favicons on left

**Comment by zantyru** [comment]
@PalmerAL I did execute
`sudo dpkg -i min_1.10.0_amd64.deb`
Although `dpkg` has finished with error, this has overwritten my previous version of Min browser.

Collapsable task's lists is good :^)

**Comment by PalmerAL** [comment]
Package name is fixed in 1d701e64641fbca38e8967917922a5ed6d562d16.

**Comment by dbfreem** [comment]
@PalmerAL I think with the facicons on the left the indention becomes a little less of an issue. You can easily see where one task ends and another begins. Are you thinking about leaving the check box to toggle the favicons? If so the task may seem to run together with the favicons off.

**Comment by PalmerAL** [comment]
@dbfreem That's good, I tried some different ways of indenting it or adjusting the font size, and I couldn't really find one that looks good (and having the task names line up with the tabs is nice, because it lets you quickly scan down the list to find something).

I'm probably going to remove the checkbox, and just make favicons enabled all the time.

**Comment by PalmerAL** [comment]
Then again, lining the task name up with the icons doesn't look too bad either, and maybe the scanning isn't as big of an issue... I'm not sure yet which way is better.

**Comment by BenTearzz** [comment]
I personally prefer the icons to the left of the URLs

**Comment by PalmerAL** [comment]
@BenTearzz Do you mean as opposed to this: https://user-images.githubusercontent.com/10314059/61411883-0482cb80-a8ad-11e9-955e-61309e12e3c6.png, or as opposed to having them on the right?

**Comment by BenTearzz** [comment]
I prefer it like it is in the picture you linked

**Comment by PalmerAL** [comment]
I think it's useful to have the tasks and tabs lined up, so I'm going to go with the option from https://github.com/minbrowser/min/pull/784#issuecomment-511681516 for now.

**Comment by PalmerAL** [comment]
I think everything here looks good, so I'm going to go ahead and merge this. Thanks for all your feedback on this!

Some other things that could be added in the future:

* Select multiple tabs at once (#478)
* Tab suspending (#355)
* Figure out something better to do with the task title.
* Leave tasks open even after the last tab is closed, which would make it easier to rearrange tabs (from the survey)

**Comment by PalmerAL** [comment]
I've also added a `!nametask` shortcut in d780446fb9cc1a361fbf6b8419dfca7a5d0dbdc0.
